request.go

```diff
@@ -356,8 +350,12 @@ func unmarshalNode(data *Node, model reflect.Value, included *map[string]*Node)
 				} else if v.Kind() == reflect.Int {
 					at = v.Int()
 				} else {
-					er = ErrInvalidTime
-					break
+					// Return error immediately to ensure a runtime panic doesn't swallow it.
+					return &ErrorObject{
```

Comment:
So this is a suggested response object, right?  IOW, Unmarshal will return an error which is a `*jsonapi.ErrorObject`, and what the consumer does with it is up to them, e.g. they could simply turn around and serialize it as json to the consumer or respond in an another fashion if they see fit?